DataConnectors/ProofpointPOD/ProofpointSentinelConnector/__init__.py

```diff
@@ -0,0 +1,156 @@
+            print('Error while connectiong to websocket {}'.format(err))
+            return None
+
+    def gen_chunks_to_object(self,data,chunksize=100):
+        chunk = []
+        for index, line in enumerate(data):
+            if (index % chunksize == 0 and index > 0):
+                yield chunk
+                del chunk[:]
+            chunk.append(line)
+        yield chunk
+
+    def gen_chunks(self,data,event_type):
+        for chunk in self.gen_chunks_to_object(data, chunksize=10000):
+            print(len(chunk))
+            obj_array = []
+            for row in chunk:
+                if row != None and row != '':
+                    y = json.loads(row)
+                    y.update({'event_type': event_type})
+                    #print(y)
+                    obj_array.append(y)
+            body = json.dumps(obj_array)
+            self.post_data(body,len(obj_array),event_type)
+
+    def build_signature(self, date, content_length, method, content_type, resource):
+        x_headers = 'x-ms-date:' + date
+        string_to_hash = method + "\n" + str(content_length) + "\n" + content_type + "\n" + x_headers + "\n" + resource
+        bytes_to_hash = bytes(string_to_hash, encoding="utf-8")
+        decoded_key = base64.b64decode(shared_key)
+        encoded_hash = base64.b64encode(
+            hmac.new(decoded_key, bytes_to_hash, digestmod=hashlib.sha256).digest()).decode()
+        authorization = "SharedKey {}:{}".format(customer_id, encoded_hash)
+        return authorization
+
+    def post_data(self,body,chunk_count,event_type):
+        method = 'POST'
+        content_type = 'application/json'
+        resource = '/api/logs'
+        rfc1123date = datetime.datetime.utcnow().strftime('%a, %d %b %Y %H:%M:%S GMT')
+        content_length = len(body)
+        signature = self.build_signature(rfc1123date, content_length, method, content_type,
+                                    resource)
+        uri = 'https://' + customer_id + '.ods.opinsights.azure.com' + resource + '?api-version=2016-04-01'
+        headers = {
+            'content-type': content_type,
+            'Authorization': signature,
+            'Log-Type': 'ProofpointPOD_' + event_type,
+            'x-ms-date': rfc1123date
+        }    
+        response = requests.post(uri, data=body, headers=headers)
+        if (response.status_code >= 200 and response.status_code <= 299):
+            logging.info("Chunk was processed({} events)".format(chunk_count))
+            print("Chunk was processed({} events)".format(chunk_count))
+        else:
+            print("Response code: {}".format(response.status_code))
+            logging.warn("Response code: {}".format(response.status_code))
+
+    def get_data(self, event_type=None):
+        sent_events = 0
+        ws = self.set_websocket_conn(event_type)
+        time.sleep(2)
+        print(ws)
+        if ws is not None:
+            events = []
+            while True:
```

Comment:
When will the "while" loop end? 

And ws needs to be closed after the call.

Reply:
while True -  Infinite loop waiting for WebSocket data. Loop will end after getting all messages from the socket or in case of Error exception.
It will close on its own but to on safe side we`ll add closing handling